pytensor/link/pytorch/dispatch/scalar.py

```diff
@@ -38,3 +37,24 @@ def pytorch_func(*args):
             )
 
     return pytorch_func
+
+
+@pytorch_funcify.register(ScalarLoop)
+def pytorch_funicify_ScalarLoop(op, node, **kwargs):
+    update = pytorch_funcify(op.fgraph)
+
+    def inner(steps, start, constant, update=update, is_while=op.is_while):
+        # easiest way to do it is to loop
+        c = start
+        for i in range(steps):
+            outs = update(c, constant)
+            if is_while:
+                n, done = outs
+                if done:
+                    return n
+                c = n
+            else:
+                c = outs[0]
+        return c
```

Comment:
I find the different implementation for while / not while cleaner, it can even be distinct functions.

```python
if op.is_while:
  def scalar_loop(...):
    ...
else:
  def scalar_loop(...):
    ...

return scalar_loop
```